pyrokinetics/equilibrium.py

```diff
@@ -159,6 +162,173 @@ def read_geqdsk(self, psi_n_lcfs=1.0):
 
         self.R_major = InterpolatedUnivariateSpline(psi_n, R_major)
 
+    def read_transp_cdf(self, time=-1):
+        """
+
+        Read in TRANSP netCDF and populates Equilibrium object
+
+        """
+
+        import netCDF4 as nc
+        import numpy as np
+        from scipy.interpolate import (
+            RBFInterpolator,
+            InterpolatedUnivariateSpline,
+            RectBivariateSpline,
+        )
+
+        data = nc.Dataset(self.eq_file)
+
+        nr = len(data["XB"][-1, :])
+
+        time_cdf = data["TIME3"][:]
+
+        if isinstance(time, int):
+            itime = time
+        elif isinstance(time, float):
+            itime = np.argmin(abs(time_cdf - time))
+        else:
+            raise ValueError("time input needs to be float or int")
+
+        ntheta = 256
+        theta = np.linspace(0, 2 * np.pi, ntheta)
+        theta = theta[:, np.newaxis]
+
+        # Calculate flux surfaces from moments up to 17
+        R_mom_cos = np.empty((17, ntheta, nr))
+        R_mom_sin = np.empty((17, ntheta, nr))
+        Z_mom_cos = np.empty((17, ntheta, nr))
+        Z_mom_sin = np.empty((17, ntheta, nr))
+
+        for i in range(len(R_mom_cos)):
+            try:
+                R_mom_cos[i, :, :] = (
+                    np.cos(i * theta) * data[f"RMC{i:02d}"][itime, :] * 1e-2
+                )
+            except IndexError:
+                break
+            Z_mom_cos[i, :, :] = (
+                np.cos(i * theta) * data[f"YMC{i:02d}"][itime, :] * 1e-2
+            )
+            if i == 0:
```

Comment:
I'm guessing you need this branch as TRANSP decides not to store the RMS0 data as it is automatically zero (i.e. the other branch would calculate the same data but it will error as the key to data is invalid. Worth a comment?